app/routes/candidates.js

```diff
@@ -26,10 +27,20 @@ router.get('/', function(req, res, next) {
     params.$district = req.query.district;
   }
 
-  db.each(selectClause + whereClause, params, function(err, row) {
-    candidates.push(row);
-  }, function() {
-    res.send(candidates);
+  db.serialize(function() {
+      var generalElection = '';
+	  // Find out if we only need to show General Election candidates
+	  db.get(selectClauseGenElect + whereClause, params, function(err, row) {
```

Comment:
I'm trying to find if any candidates are in the general, and if so, then only select those candidates to show. It was more convoluted to try and do a single query. Of course you might figure out a more elegant way.